consensus/consensus.go

```diff
@@ -585,11 +622,68 @@ func (consensus *Consensus) Prepare(chain consensus_engine.ChainReader, header *
 	return nil
 }
 
-// AccumulateRewards credits the coinbase of the given block with the mining
+// accumulateRewards credits the coinbase of the given block with the mining
 // reward. The total reward consists of the static block reward and rewards for
 // included uncles. The coinbase of each uncle block is also rewarded.
-func accumulateRewards(config *params.ChainConfig, state *state.DB, header *types.Header) {
-	// TODO: implement mining rewards
+func (consensus *Consensus) accumulateRewards(
+	config *params.ChainConfig, state *state.DB, header *types.Header,
+) {
+	if header.ParentHash == (common.Hash{}) {
+		// This block is a genesis block,
+		// without a parent block whose signer to reward.
+		return
+	}
+	if consensus.ChainReader == nil {
+		utils.GetLogInstance().Error("accumulateRewards: ChainReader is nil")
+		return
+	}
+	parent := consensus.ChainReader.GetBlockByHash(header.ParentHash)
+	if parent == nil {
+		utils.GetLogInstance().Error(
```

Comment:
please just return error in this function

Reply:
This isn't that frequent.  Perhaps will happen once for genesis block, but shouldn't happen anymore after that.  If we see this message, it means that the current block is on chain but the previous block is not, which is a serious invariant violation: We deserve to be loud about that failure.